pkg/lib/cockpit-components-truncate.jsx

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+
+const minWidthCharacters = 16;
+
+const sliceContent = (str, slice) => [
+    str.slice(0, str.length - slice),
+    str.slice(-slice),
+];
+
+export const Truncate = ({
```

Comment:
Oh, this implements all of PF Truncate features! :-) Just position="middle" would have been enough.

Reply:
We need a copyright notice in this file like
```
/*
 * This file is part of Cockpit.
 *
 * Copyright (C) 2024 Mohd Mohsin
 *
 * Cockpit is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation; either version 2.1 of the License, or
 * (at your option) any later version.
 *
 * Cockpit is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 */
```

Reply:
Can this also explain why it exists when we have `Truncate` from PF already.

Reply:
Yes, and since I am the one complaining, I have added it. Thanks for pointing this out!

Reply:
Thanks and thanks for making the PF issue